apps/opencs/model/world/nestedcoladapterimp.cpp

```diff
@@ -1069,23 +1070,66 @@ namespace CSMWorld
         switch (subColIndex)
         {
             case 0: return isInterior;
-            case 1: return (isInterior && !behaveLikeExterior) ?
-                    cell.mAmbi.mAmbient : QVariant(QVariant::UserType);
-            case 2: return (isInterior && !behaveLikeExterior) ?
-                    cell.mAmbi.mSunlight : QVariant(QVariant::UserType);
-            case 3: return (isInterior && !behaveLikeExterior) ?
-                    cell.mAmbi.mFog : QVariant(QVariant::UserType);
-            case 4: return (isInterior && !behaveLikeExterior) ?
-                    cell.mAmbi.mFogDensity : QVariant(QVariant::UserType);
+            case 1:
+            {
+                if (isInterior && !behaveLikeExterior)
+                    return cell.mAmbi.mAmbient;
+                else
+                {
+                    UserInt i(cell.mAmbi.mAmbient);
+                    return QVariant(QVariant::fromValue(i));
+                }
```

Comment:
I think this is less invasive than returning a struct with a bool and an int, for example.  Such a change will mean a significant amount of change elsewhere in OpenCS.